Jenkinsfile

```diff
@@ -0,0 +1,81 @@
+                        error("Database has already been updated today.")
+                    }
+                }
+
+                // Start the update from the latest version of the database
+                sh '''cp ${DB_LINK} ${DB}
+                      chmod 640 ${DB}'''
+
+                // Try to update the database. Because the PBS job has a time limit, the update can fail for two different reasons:
+                //  1. the time limit was reached
+                //  2. an error occured
+                // In the first case, we don't want to mark the build as failed, because the database will still be usable.
+                script {
+                    catchError(buildResult: 'UNSTABLE', stageResult: 'UNSTABLE') {
+                       sh "qsub -v DB build_master_index"
+                    }
+                    if (currentBuild.result == 'UNSTABLE') {
+                        // Check if the job hit the walltime. Mark the build as successful if that is the case.
+                        timeout = """${sh(returnStdout: true, script: 'grep -c "PBS: job killed: walltime [0-9]* exceeded limit" log.out')}""".trim()
+                        if ("$timeout"  == "1") {
+                            echo "Database update timed out."
+                            currentBuild.result == 'SUCCESS'
+                        } else {
+                            error "Error updating database."
+                        }
+                    }
+                }
+
+                sh "chmod 440 ${DB}"
+            }
+        }
+    }
+    post {
+        success {
+            // Update was successful, so we can now update the symlink to point to this new version of the database
+            sh "ln -sf ${DB} ${DB_LINK}"
+            // Prune old versions of the database that have not been accessed in the last 14 days.
+            sh 'find ${DB_PATH}/daily -type f -atime +14 -exec rm -fv {} \\;'
+        }
+        unstable {
+            // The update timed out, but since no other errors have occured, this database should be usable and more complete than the previous version.
+            sh "ln -sf ${DB} ${DB_LINK}"
+            // Prune old versions of the database that have not been accessed in the last 14 days.
+            sh 'find ${DB_PATH}/daily -type f -atime +14 -exec rm -fv {} \\;'
+        }
+        failure {
+            // Delete the failed attempt to update the database
+            sh "rm -f ${DB}"
+        }
+        aborted {
+            // Delete the new database, as we cannot be sure about the state of the database in that case
+            sh "rm -f ${DB}"
+        }
+        unsuccessful {
+            emailext (
+                body: "${currentBuild.currentResult}: Job ${env.JOB_NAME} build ${env.BUILD_NUMBER}\n More info at: ${env.BUILD_URL}",
+                subject: "Jenkins Build ${currentBuild.currentResult}: Job ${env.JOB_NAME}",
+                to: 'micael.oliveira@anu.edu.au'
```

Comment:
I think it is always best for more than one person to get these warnings, in case you're away.

Reply:
Agreed. Who should I add here?

Reply:
add me if you like